sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -56,139 +56,12 @@ Import-Module Az.Resources
-  Write-Host ""
-
-  Write-Host "$($indent)-AzureRegion`t`tThe Azure region that resources should be created in.  This value should be"
-  Write-Host "$($indent)the name of the region, in lowercase, with no spaces.  For example: southcentralus"
-  Write-Host ""
-  Write-Host "$($indent)`t`t`t`tDefault: South Central US (southcentralus)"
-  Write-Host ""  
-}
-
-function SelectRandomCharacters
-{    
-    <#
-      .SYNOPSIS
-        Selects a number of random characters from a set.
-    #>
-
-    param
-    (
-        [Parameter(Mandatory=$true)]
-        [int]$length, 
-
-        [Parameter(Mandatory=$true)]
-        [string]$characters
-    ) 
-
-    $random = (1..$length | ForEach { Get-Random -Maximum $characters.length })
-    return (-join $characters[$random])
-}
-
-function GenerateRandomPassword()
-{
-    <#
-      .SYNOPSIS
-        Generates a random password that can be assigned to a service principal.
-        
-      .DESCRIPTION
-        The password generated by this function will contain a mix of alpha,
-        numeric, and special characters and will vary a small amount in its length.        
-    #>
-
-    $baseLength = (Get-Random -Minimum 28 -Maximum 37)      
-    $upper = (Get-Random -Minimum 6 -Maximum ([int][Math]::Ceiling($baseLength / 3)))  
-    $special = (Get-Random -Minimum 2 -Maximum 3)     
-    $lower = ($baseLength - $upper - $special - $special)                                 
-
-    $password = SelectRandomCharacters $lower "abcdefghiklmnoprstuvwxyz"
-    $password += SelectRandomCharacters $upper "ABCDEFGHKLMNOPRSTUVWXYZ"
-    $password += SelectRandomCharacters $special "1234567890"
-    $password += SelectRandomCharacters $special "!$%&/()=?}][{@#*+"
-
-    $scrambled = ($password.ToCharArray()) | Get-Random -Count ($password.Length)
-    return (-join $scrambled)
-}
-
-function TearDownResources
-{
-    <#
-      .SYNOPSIS
-        Cleans up any Azure resources created by the script.
-        
-      .DESCRIPTION
-        Responsible for cleaning up any Azure resources created 
-        by the script in case of failure.
-    #>
-    
-    param
-    (
-        [Parameter(Mandatory=$true)]
-        [bool] $cleanResourceGroup
-    )
-    
-    Write-Host("Cleaning up resources that were created:")
-    
-    if ($cleanResourceGroup)
-    {
-        try 
-        {
-            Write-Host "`t...Removing resource group `"$($ResourceGroupName)`""
-            Remove-AzResourceGroup -Name "$($ResourceGroupName)" -Force | Out-Null
-        }
-        catch 
-        {
-            Write-Error "The resource group: $($ResourceGroupName) could not be removed.  You will need to delete this manually."
-            Write-Error ""            
-            Write-Error $_.Exception.Message
-        }
-    }
-}
+. .\functions\validation-functions.ps1
+. .\functions\help-functions.ps1
+. .\functions\credentials-functions.ps1
+. .\functions\eventhubs-functions.ps1

```

Comment:
I'd say that we should consider doing a "look locally for the include but if not found execute from the repository", but that may be more trouble than its worth and/or may make people more wary about running due to potentially executing remote code.

Thoughts?

Something similar to this is what I had in mind:

```powershell
if (Test-Path ".\functions\help-functions.ps1" -PathType Leaf)
{
    .\functions\help-functions.ps1
}
else
{
    iex ((New-Object System.Net.WebClient).DownloadString('https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs/assets/functions/help-functions.ps1'))
}
```

Reply:
> "look locally for the include but if not found execute from the repository"

Interesting, I admit lacking a bit of experience in this. Why would the functions miss in first place?

Reply:
I was thinking specifically of the case where someone grabbed the main script rather than cloning the repository.   The scenario of  `I just want to run samples, why do I have to clone things!`.    That said, I'd definitely leave it as-is for the time being.  It's something that we can tack on in the future if there turns out to be a need.

Reply:
I think your point is very valid, are you sure you do not want us to try tackling this issue?

Reply:
Pretty sure that we could come up with a bunch of corner cases and improvements.   However, this is a script that folks are likely to run once or very infrequently.  I'd hesitate to put too much engineering into trying to tune the experience without feedback.   If we let folks try it out and they don't like it, I expect that we'll see some issues opened.